homeassistant/components/tesla_wall_connector/config_flow.py

```diff
@@ -0,0 +1,160 @@
+"""Config flow for Tesla Wall Connector integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from tesla_wall_connector import WallConnector
+from tesla_wall_connector.exceptions import WallConnectorError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.dhcp import IP_ADDRESS
+from homeassistant.const import CONF_HOST, CONF_SCAN_INTERVAL
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import (
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    WALLCONNECTOR_DEVICE_NAME,
+    WALLCONNECTOR_SERIAL_NUMBER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    wall_connector = WallConnector(
+        host=data[CONF_HOST], session=async_get_clientsession(hass)
+    )
+
+    version = await wall_connector.async_get_version()
+
+    return {
+        "title": WALLCONNECTOR_DEVICE_NAME,
+        WALLCONNECTOR_SERIAL_NUMBER: version.serial_number,
+    }
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Tesla Wall Connector."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize config flow."""
+        super().__init__()
+        self.ip_address = None
+        self.serial_number = None
+
+    async def async_step_dhcp(self, discovery_info) -> FlowResult:
+        """Handle dhcp discovery."""
+        self.ip_address = discovery_info[IP_ADDRESS]
+        _LOGGER.info("Discovered Tesla Wall Connector at [%s]", self.ip_address)
```

Comment:
```suggestion
        _LOGGER.debug("Discovered Tesla Wall Connector at [%s]", self.ip_address)
```

Max debug.  We don't want to log every time dhcp discovery runs

Reply:
Fixed in a7f0a65